nautobot/cloud/views.py

```diff
@@ -67,50 +71,130 @@ class CloudNetworkUIViewSet(NautobotUIViewSet):
+                url_name="cloud:cloudnetwork_prefixes",
+                related_object_attribute="prefixes",
+            ),
+            object_detail.DistinctViewTab(
+                weight=900,
+                tab_id="circuits",
+                label="Circuits",
+                url_name="cloud:cloudnetwork_circuits",
+                related_object_attribute="circuit_terminations",
+            ),
+            object_detail.DistinctViewTab(
+                weight=1000,
+                tab_id="cloud_services",
+                label="Cloud Services",
+                url_name="cloud:cloudnetwork_cloud_services",
+                related_object_attribute="cloud_services",
+            ),
+        ),
+    )
+
+    def get_extra_context(self, request, instance):
         context = super().get_extra_context(request, instance)
-        if self.action == "retrieve":
-            prefixes = instance.prefixes.restrict(request.user, "view")
-            prefixes_table = PrefixTable(prefixes.select_related("namespace"), hide_hierarchy_ui=True)
-            prefixes_table.columns.hide("location_count")
-            prefixes_table.columns.hide("vlan")
-
-            children = instance.children.restrict(request.user, "view")
-            children_table = CloudNetworkTable(children)
-            children_table.columns.hide("parent")
-
-            circuits = Circuit.objects.restrict(request.user, "view").filter(
-                Q(circuit_termination_a__cloud_network=instance.pk)
-                | Q(circuit_termination_z__cloud_network=instance.pk)
-            )
-            circuits_table = CircuitTable(circuits)
-            circuits_table.columns.hide("circuit_termination_a")
-            circuits_table.columns.hide("circuit_termination_z")
+        context.update({"object_detail_content": self.object_detail_content})
+        return context
 
-            cloud_services = instance.cloud_services.restrict(request.user, "view")
-            cloud_services_table = CloudServiceTable(cloud_services)
-            cloud_services_table.columns.hide("cloud_network_count")
+    @action(detail=True, url_path="children")
+    def children(self, request, *args, **kwargs):
+        instance = self.get_object()
+        children = instance.children.restrict(request.user, "view")
+        children_table = CloudNetworkTable(
+            data=children,
+            extra_columns=[("actions", ButtonsColumn(model=CloudNetwork, return_url_extra="?tab=children"))],
+        )
+        children_table.columns.hide("parent")
+        RequestConfig(
+            request, paginate={"paginator_class": EnhancedPaginator, "per_page": get_paginate_count(request)}
+        ).configure(children_table)
+        child_url = reverse("cloud:cloudnetwork_add")
+        child_table_add_url = f"{child_url}?parent={instance.id}"
+        return Response(
+            {"children_table": children_table, "active_tab": "children", "child_table_add_url": child_table_add_url}
+        )
```

Comment:
This *feels* like a step backwards to me. Ideally each of these distinct detail action views would be able to have their own `object_detail_content` that just contains a single `ObjectsTablePanel`, instead of needing all of this custom data logic in each case. Can we investigate that option?

Reply:
https://github.com/nautobot/nautobot/pull/7158/commits/175c238282cd6bf1cd91dd04884dc390599d2b0f